test/util/spec.helpers.js

```diff
@@ -0,0 +1,117 @@
+                    },
+                    '/step3': {
+                        next: '/step4'
+                    },
+                    '/step3a': {
+                        next: '/step4'
+                    },
+                    '/step4': {}
+                };
+            });
+
+            it('should return all steps that have /step4 as next', function () {
+                getPreviousSteps(steps, '/step4').should.be.eql(['/step3', '/step3a']);
+            });
+
+            it('should return an empty array if step isn\'t linked to', function () {
+                getPreviousSteps(steps, '/step5').should.be.an('array')
+                  .and.have.property('length')
+                  .and.equal(0);
+            });
+        });
+
+        describe('using forks', function () {
+            beforeEach(function () {
+              steps = {
+                  '/step1': {
+                      forks: [{
+                        target: '/step2'
+                      }]
+                  },
+                  '/step2': {
+                      forks: [{
+                        target: '/step3'
+                      }, {
+                        target: '/step4'
+                      }]
+                  },
+                  '/step3': {
+                      forks: [{
+                        target: '/step3a'
+                      }]
+                  },
+                  '/step3a': {
+                      forks: [{
+                        target: '/step4'
+                      }]
+                  },
+                  '/step4': {}
+              };
+            });
+
+            it('should return a step that has route as a fork target', function () {
+                getPreviousSteps(steps, '/step2').should.be.eql(['/step1']);
+            })
+
+            it('should return all steps that have given step as fork target', function () {
+                getPreviousSteps(steps, '/step4').should.be.eql(['/step2', '/step3a']);
+            });
+        });
+
+        describe('using next property and forks', function () {
+            beforeEach(function () {
+                steps = {
+                    '/step1': {
+                        next: '/step2',
+                        forks: [{
+                          target: '/step3'
+                        }]
+                    },
+                    '/step2': {
+                        next: '/step3',
+                        forks: [{
+                          target: '/step3a'
+                        }, {
+                          target: '/step4'
+                        }]
+                    },
+                    '/step3': {},
+                    '/step3a': {},
+                    '/step4': {}
+                }
+            });
+
+            it('should return all steps that link to route', function () {
+                getPreviousSteps(steps, '/step3').should.be.eql(['/step1', '/step2']);
+            });
+        });
+    });
```

Comment:
I guess the next only/forks only/next and forks `describe`s are there to show they can work independently but I'm wondering if there's a better way to show this? There's no configurable way to just return `next`s or `forks` (which is a good thing) so could the `it`s say something like the `describe`s using a realistic `steps` object? Maybe.